app/pages/resource/ResourcePage.js

```diff
@@ -263,13 +263,20 @@ class UnconnectedResourcePage extends Component {
                         <ResourcePanel header={t('ResourceInfo.reserveTitle')}>
                           <>
                             {resource.externalReservationUrl && (
-                              <form action={resource.externalReservationUrl} method="post" target="_blank">
-                                <input
-                                  className="btn btn-primary"
-                                  type="submit"
-                                  value="Siirry ulkoiseen ajanvarauskalenteriin"
-                                />
-                              </form>
+                              <div>
+                                <div className="app-ResourcePage__external_reservation_text">
+                                  <span>
+                                    {resource.externalReservationText}
+                                  </span>
+                                </div>
+                                <form action={resource.externalReservationUrl} method="post" target="_blank">
+                                  <input
+                                    className="btn btn-primary"
+                                    type="submit"
+                                    value="Siirry ulkoiseen ajanvarauskalenteriin"
```

Comment:
Will this always be in Finnish visible to the client, or am I mistaken and rusty in react?

Reply:
This will always be in Finnish.  Tampere city wants everything in Finnish now, they are taking off the other language selection for now. So may be when we can add the translation for these and other when they decide to have the site in other language too.

Reply:
Check!

As a side note, is there any "string discovery" implemented in the project, that extracts strings into the json translation files? To hunt for untranslated strings among translated ones can become a cumbersome task the more an application is developed and more strings (with more elaborate cases) are added.

I wouldn't go as far as mandate a translation here but just a friendly reminder that it is easier to gather strings during development rather than as a task at a later point. :smiley: